src/VisualStudio/Core/Def/Implementation/VirtualMemoryNotificationListener.cs

```diff
@@ -21,6 +22,9 @@ namespace Microsoft.VisualStudio.LanguageServices
     [Export, Shared]
     internal sealed class VirtualMemoryNotificationListener : ForegroundThreadAffinitizedObject, IVsBroadcastMessageEvents
     {
+        // memory threshold to turn off full solution analysis - 200MB
+        private const long MemoryThreshold = 200 * 1024 * 1024;
```

Comment:
in 64bit machine, current 500mb limit seems fine (since max is 4GB), but it looks like (by looking at shell code), the event doesnt care about it. always raise it on 500mb mem left.